test/http_to_do_test.exs

```diff
@@ -10,18 +10,26 @@ defmodule HTTPServerTest.ToDo do
              )
 
   setup do
-    act_arrange_test_todo = JSON.encode!(%{"1" => %{"todo1" => "Act"}, "2" => %{"todo2" => "Arrange"}})
-    File.write!(@file_path, act_arrange_test_todo)
+    act_arrange_test_todo =
+      JSON.encode!(%{
+        "1" => %{"todo1" => "Act"},
+        "2" => %{"todo2" => "Arrange"},
+        "4" => %{"todo4" => "Teardown"}
+      })
+
+    File.write(@file_path, act_arrange_test_todo)
+    :ok
   end
 
-  test "Can write \"{\"todo3\":\"Assert\"}\" to data file without overwriting previous data" do
+  test "Can add valid ToDo \"{\"todo3\":\"Assert\"}\"" do
```

Comment:
These test names look a lot better, but you really do not need those implementation details  in the name.

```suggestion
  test "Can add valid ToDo" do
```

and in other places:

```elixir
  test "Can delete valid ToDo" do
  ...
```

```elixir
  test "Cannot delete a ToDo with an id that doesn't exist" do
  ...
```

```elixir
  test "Can only delete a ToDo with an id that is malformed" do
  ...
```